### ISSUE TITLE
iOS+Android touch controls

### ISSUE THREAD

**Issue by hhas** [yaelatletl/PathsOfJjaro#14]
Create proof-of-concept designs for touchscreen controls on iOS and Android to determine the best control layout(s).

### Motivation

To differentiate MCR from Bungie’s new Marathon game (Steam + Console), MCR should target Mobile (Android, iOS, Switch) as our primary market. (While we could release on PC too, through Steam or GOG, we’d only do this if Bungie requested it.) While Switch has its own physical controls, iOS and Android are limited to touchscreen input by default so our touch controls must work really well.

### Problem

With dual virtual joysticks (two-handed grip), the user is limited to performing all actions using two thumbs only, making it difficult to perform move and fire actions simultaneously. 

On pressure-sensitive devices, firing could be performed by pressing hard on a virtual joystick—this is likely the best solution where available (all iOS devices, X% of Android devices).

On non-pressure-sensitive devices, one solution might be to swap sidestep and turn axes, so the “move” stick operates tank-style and the “look” stick performs sidestepping and look up/look down, with the primary trigger positioned under the look stick. That allows greater movement control when one thumb is held on the primary firing trigger for full auto firing (typically using AR or flamethrower). Settings could include automatic look up/down option, where the Player automatically adjusts vertical angle to target the closest enemy directly ahead, reducing reliance on the “look” joystick.

An alternate touch control mode might be provided where the Player is controlled by “tap and drag” finger movements: tapping at a floor position moves to it, dragging turns the player, tapping on an enemy fires at it, etc.

More research is needed. This includes gathering stats on the percentage of customers using pressure-sensitive vs non-pressure-sensitive Android devices.

### Notes

We can reduce the total number of touch buttons by combining Fire and Action so that Fire control[s] automatically switch behavior to Action when the Player is looking directly at, and in reach of, an Action-able surface. e.g. Wheeels! does this, combining the M2 version’s separate Fire and Action keys into a single Action button that performs one or the other dependent on context, with distinct throw and push “hand” gestures indicating which will be performed when Action is pressed.

While Inventory and Automap overlays will require their own buttons to display, these screen could auto-dismiss after, say, 5 seconds, so that the user does not need to explicitly dismiss them every time, Inventory and Terminal screens could also auto-dismiss immediately upon the user touching a movement control.

Overlay interactions such as scrolling terminals and zooming maps can be performed using standard mobile drag and pinch gestures so do not require virtual buttons on touchscreen. These operations might also be automatable, e.g. Terminal text could scroll at reading speed or, when multi-page text is dragged up slowly and released, continue scrolling at that speed until [e.g.] a map view screen appears, at which point it stops (since map screens take reader longer to digest) until touched again; Automap could appear zoomed out (provides context), auto-zoom in on player for a few seconds (provides detail), then zoom back out before auto-dismissing.

Other ### discussions

https://www.reddit.com/r/gamedev/comments/pozh3x/what_were_the_best_touchscreen_controls_youve/

https://gamesperience.wordpress.com/2018/04/06/mobile-shooters-the-quest-for-the-perfect-touch-control-layout/

https://www.eurogamer.net/ex-battlefield-developer-ben-cousins-unveils-the-drowning-the-iphone-and-ipad-fps-you-can-play-with-just-two-fingers-on-one-hand

**Comment by yaelatletl** [comment]
I strongly disagree about our target, I know differentiation is crucial, but in this case focusing mainly on mobile will hinder our ability to fully realize a technically upgraded remake. We can take mobile platforms into consideration, but not have them as a focus. Otherwise we would be creating a more immersive world that cannot be enjoyed to its full extent.
